responsibleai_vision/responsibleai_vision/utils/image_reader.py

```diff
@@ -6,13 +6,36 @@
 import base64
 from io import BytesIO
 from typing import Any, Tuple, Union
+from urllib.parse import urlparse
 
 import requests
 from numpy import asarray
 from PIL import Image
+from requests.adapters import HTTPAdapter, Retry
 
-from responsibleai_vision.common.constants import (AutoMLImagesModelIdentifier,
-                                                   CommonTags)
+from responsibleai_vision.common.constants import (
+    AutoMLImagesModelIdentifier,
+    CommonTags,
+)
+
+# domain mapped session for reuse
+_requests_sessions = {}
```

Comment:
if there are a ton of images won't this list become very large?  not sure how big this requests.Session() object is.  why not just use a single session for all of the images?  As far as I can tell you can use it for all url get calls:
https://requests.readthedocs.io/en/latest/_modules/requests/sessions/
perhaps we can just remove this "_requests_sessions = {}" dictionary altogether?